scripts/pipeline_generator/step.py

```diff
@@ -1,9 +1,72 @@
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, root_validator, model_validator
 from typing import List, Dict, Any, Optional
-
-from .utils import AgentQueue
+from typing_extensions import Self
+from .utils import AgentQueue, GPUType
```

Comment:
keep the separate line between local and third-party imports?

Reply:
done